### ISSUE TITLE
'Space Around Operator' rule not working

### ISSUE THREAD

**Issue by seeliang** [sasstools/sass-lint#920]
What version of Sass Lint are you using?
1.9.1
Please include any relevant parts of your configuration
space-around-operator
What did you do? Please include the actual source code causing the issue.

``` sass
  @mixin font-weight($weight: medium) {

  @if $weight==light {
    font-weight: $font-weight-1;
  }
}
```

What did you expect to happen?
a error with 'Space Around Operator'

What actually happened? Please include any error messages given to you by Sass Lint.
no error

If you're using a IDE plugin have you tried the CLI too?
Atom

**Comment by DanPurdy** [comment]
is it the `$font-weight-1` you're talking about here? If so it's because technically that's a valid variable name in Sass. If you were to use any other operator there it would ('should') flag up as an error but because you've not spaced it all near a variable it assumes it's the variable name. There's a similar discussion in another issue which I don't seem to be able to find right now...

There's an argument for checking every variable within this rule to to raise issue with variable names like this but then we can't reliably say that it wasn't the developers intention to name it this way. I'll mark it as invalid for now until I hear whether this is what you meant.

**Comment by seeliang** [comment]
Sorry my bad, not a very good sample

it's the 
``` sass
 @if $weight==light
```
part

**Comment by DanPurdy** [comment]
Ah cool, I wasn't sure, just because i'd seen a similar issue recently. 

Ok yeah, that's definitely a bug. Will have a look at it, as usual PR's welcome!

**Comment by bgriffith** [comment]
I can't replicate this, I'm getting an warning. Did you replicate this issue @DanPurdy ?

**Comment by DanPurdy** [comment]
Nope, hadn't had time yet. So you are seeing an error @bgriffith

**Comment by bgriffith** [comment]
Hey @seeliang - Could you post your config file? I can't replicate the issue and we have tests that are passing for this example.

**Comment by seeliang** [comment]
Sure,
my settings 
 
package.json
```json
    "gulp": "3.9.1",
    "gulp-sass-lint": "1.2.0",
```
.sass-lint.yml
```yml
 # sass-lint config generated by make-sass-lint-config v0.1.1
#
# The following scss-lint Linters are not yet supported by sass-lint:
# DisableLinterReason, ElsePlacement, PropertyCount, SelectorDepth
# SpaceAroundOperator, TrailingWhitespace, UnnecessaryParentReference, Compass::*
#
# The following settings/values are unsupported by sass-lint:
# Linter Indentation, option "allow_non_nested_indentation"
# Linter Indentation, option "character"
# Linter NestingDepth, option "ignore_parent_selectors"
# Linter PropertySortOrder, option "min_properties"
# Linter PropertySortOrder, option "separate_groups"
# Linter SpaceBeforeBrace, option "allow_single_line_padding"
# Linter VendorPrefix, option "identifier_list"

options:
  formatter: stylish
  merge-default-rules: false
rules:
  bem-depth:
    - 0
    - max-depth: 1
  border-zero:
    - 1
    - convention: 0
  brace-style:
    - 1
    - allow-single-line: true
  class-name-format:
    - 1
    - convention: hyphenatedbem
  clean-import-paths:
    - 1
    - filename-extension: false
      leading-underscore: false
  empty-args: 1
  empty-line-between-blocks:
    - 1
    - ignore-single-line-rulesets: true
  extends-before-declarations: 1
  extends-before-mixins: 1
  final-newline:
    - 1
    - include: true
  force-attribute-nesting: 1
  force-element-nesting: 1
  force-pseudo-nesting: 1
  function-name-format:
    - 1
    - allow-leading-underscore: true
      convention: hyphenatedlowercase
  hex-length:
    - 1
    - style: short
  hex-notation:
    - 1
    - style: lowercase
  id-name-format:
    - 0
    - convention: hyphenatedlowercase
  indentation:
    - 1
    - size: 2
  leading-zero:
    - 1
    - include: false
  mixin-name-format:
    - 1
    - allow-leading-underscore: true
      convention: hyphenatedlowercase
  mixins-before-declarations: 1
  nesting-depth:
    - 1
    - max-depth: 3
  no-color-keywords: 1
  no-color-literals: 1
  no-combinators: 1
  no-css-comments: 1
  no-debug: 1
  no-duplicate-properties: 1
  no-empty-rulesets: 1
  no-extends: 0
  no-ids: 1
  no-important: 1
  no-invalid-hex: 1
  no-mergeable-selectors: 1
  no-misspelled-properties:
    - 1
    - extra-properties: []
  no-qualifying-elements:
    - 1
    - allow-element-with-attribute: false
      allow-element-with-class: false
      allow-element-with-id: false
  no-trailing-whitespace: 1
  no-trailing-zero: 1
  no-transition-all: 0
  no-url-protocols: 0
  no-vendor-prefixes:
    - 1
    - additional-identifiers: []
      excluded-identifiers: []
  placeholder-in-extend: 1
  placeholder-name-format:
    - 0
    - convention: hyphenatedlowercase
  property-sort-order:
    - 1
    - ignore-custom-properties: false
  property-units:
    - 1
    - global:
        - ch
        - em
        - ex
        - rem
        - cm
        - in
        - mm
        - pc
        - pt
        - px
        - q
        - vh
        - vw
        - vmin
        - vmax
        - deg
        - grad
        - rad
        - turn
        - ms
        - s
        - Hz
        - kHz
        - dpi
        - dpcm
        - dppx
        - '%'
      per-property: {}
  quotes:
    - 1
    - style: single
  shorthand-values:
    - 1
    - allowed-shorthands:
        - 1
        - 2
        - 3
  single-line-per-selector: 1
  space-after-bang:
    - 1
    - include: false
  space-after-colon:
    - 1
    - include: true
  space-after-comma:
    - 1
    - include: true
  space-before-bang:
    - 1
    - include: true
  space-before-brace:
    - 1
    - include: true
  space-before-colon: 1
  space-between-parens:
    - 1
    - include: false
  trailing-semicolon: 1
  url-quotes: 1
  variable-for-property:
    - 0
    - properties: []
  variable-name-format:
    - 1
    - allow-leading-underscore: true
      convention: hyphenatedlowercase
  zero-unit: 1
```

**Comment by DanPurdy** [comment]
@seeliang I may be being completely blind but you don't have the rule enabled in your config..

**Comment by seeliang** [comment]
Oh my, 
i think i have being missing this set since I updated from .scss-lint to .sass-lint 

i am so so sorry about this @DanPurdy and @bgriffith

**Comment by DanPurdy** [comment]
Ha, dont worry about it @seeliang one of those things we all do at some point! 👍

**Comment by seeliang** [comment]
thx @DanPurdy
